[Config] [Configuration] Emphasize env vars purpose (#18850)
symfony/symfony-docs: configuration.rst

not impact performance) so you can change the application behavior without having
to clear the cache.

> .. caution::

Reviewer: is it a caution or more like a note ?

Author: Changed to `note`, you're right

    Environment variables are designed to store information that can
    dynamically change in a production environment, such as an API
    key that may have an expiration date. This information could be updated

>     without having to redeploy the application nor clear the cache. Thus, not

Reviewer: ```suggestion
    without having to redeploy the application. Thus, not
```

it is already document in the paragraph above `without having to clear the cache.`

Author: Removed :+1:

    everything is a good candidate for env vars, and

>     :ref:`parameters <configuration-parameters>` should be used in other cases.

Reviewer: ```suggestion
    :ref:`parameters <configuration-parameters>` should be challenged with environment configs.
```

Author: I don't know, I'm always afraid with the `this should be challenged` wording. I always have the feeling it is very vague and doesn't help to make a decision. Does anyone else have thoughts on this?

Reviewer: oki got your point now
